eth/protocols/eth/handlers.go

```diff
@@ -183,16 +187,18 @@ func AnswerGetReceiptsQuery(br services.FullBlockReader, db kv.Tx, query GetRece
 		if b == nil {
 			return nil, nil
 		}
-		results := rawdb.ReadReceipts(db, b, s)
-		if results == nil {
-			header, err := rawdb.ReadHeaderByHash(db, hash)
-			if err != nil {
-				return nil, err
-			}
-			if header == nil || header.ReceiptHash != types.EmptyRootHash {
```

Comment:
seems this optimization lost. maybe move it inside generator

Reply:
done but not in generator because it's query response optimization, not receipt